src/p2p/net_peerlist.h

```diff
@@ -285,8 +285,8 @@ namespace nodetool
     {
       if(!vl.last_seen)
         continue;
-      bs_head.push_back(vl);      
-      if(cnt++ > depth)
+      bs_head.push_back(vl);
+      if(++cnt >= depth)
```

Comment:
Assuming depth is the number of items to pick (which isn't super obvious, but appears to be the most plausible interpretation), this is still wrong, I'd do:

if (cnt++ >= depth) break;
bs_head.push_back(vl);

A depth of 0 might not be super useful though, but still.